WellLog.Lib/Helpers/ByteHelpers.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+
+namespace WellLog.Lib.Helpers
+{
+    public static class ByteHelpers
+    {
+        public static bool GetBit(this byte b, byte mask)
+        {
+            return (b & mask) > 0;
+        }
+
+        public static byte SetBit(this byte b, byte mask, bool value)
```

Comment:
"Set" in the context of OOP languages typically means you are mutating something using a reference. I don't think you should call this `SetBit` at the least don't use `Set` because that would imply the reference is being changed.

After thinking of what could be a good name for this function it looks like it's actually doing two different things.

- The first one is OR'ing the mask against the original byte
- The second one is AND'ing against the inverted mask

What does "value" mean in this context?

I think this should be split into two separate functions that names describe that the behavior is.

Reply:
the intent is to return a new byte with bit identified by the bit mask set to either a 1 or a 0 depending on the `value` passed in.

OR'ing is used to set the bit within the byte to a 1 while not changing the values of the other bits.

AND'ing the inverse is used to set the value of the bit to a 0 while not changing the values of the other bits.

I agree that the name could be more clear.  Suggestions?

Reply:
I think create two separate functions without a flag to have two different behaviors. I don't have enough knowledge on the domain to come up with good names since I don't know the intent. But I think it'll make it easier to come up with a well named function when the function has one responsibility.

Typically passing in a boolean to have a function do a different behavior is a smell because it violates single responsibility principal.

Reply:
In binary, "Set" means to change a bit to 1.  "Reset" means to change it to 0.  I would stay away from having to pass so many values into that simple function.  I have put some code below that will accomplish this task.